swig/include/gdalconst.i

```diff
@@ -228,26 +237,50 @@
 #define GDAL_DMD_HELPTOPIC "DMD_HELPTOPIC"
 #define GDAL_DMD_MIMETYPE "DMD_MIMETYPE"
 #define GDAL_DMD_EXTENSION "DMD_EXTENSION"
-#define GDAL_DMD_EXTENSIONS "DMD_EXTENSIONS"
 #define DMD_CONNECTION_PREFIX  "DMD_CONNECTION_PREFIX"
+#define GDAL_DMD_EXTENSIONS "DMD_EXTENSIONS"
 #define GDAL_DMD_CREATIONOPTIONLIST "DMD_CREATIONOPTIONLIST"
+#define GDAL_DMD_MULTIDIM_DATASET_CREATIONOPTIONLIST "DMD_MULTIDIM_DATASET_CREATIONOPTIONLIST"
+#define GDAL_DMD_MULTIDIM_GROUP_CREATIONOPTIONLIST "DMD_MULTIDIM_GROUP_CREATIONOPTIONLIST"
+#define GDAL_DMD_MULTIDIM_DIMENSION_CREATIONOPTIONLIST "DMD_MULTIDIM_DIMENSION_CREATIONOPTIONLIST"
+#define GDAL_DMD_MULTIDIM_ARRAY_CREATIONOPTIONLIST "DMD_MULTIDIM_ARRAY_CREATIONOPTIONLIST"
+#define GDAL_DMD_MULTIDIM_ATTRIBUTE_CREATIONOPTIONLIST "DMD_MULTIDIM_ATTRIBUTE_CREATIONOPTIONLIST"
+#define GDAL_DMD_OPENOPTIONLIST "DMD_OPENOPTIONLIST"
 #define GDAL_DMD_CREATIONDATATYPES "DMD_CREATIONDATATYPES"
 #define GDAL_DMD_CREATIONFIELDDATATYPES "DMD_CREATIONFIELDDATATYPES"
+#define GDAL_DMD_CREATIONFIELDDATASUBTYPES "DMD_CREATIONFIELDDATASUBTYPES"
 #define GDAL_DMD_SUBDATASETS "DMD_SUBDATASETS"
 #define GDAL_DMD_CREATION_FIELD_DOMAIN_TYPES "DMD_CREATION_FIELD_DOMAIN_TYPES"
 #define GDAL_DMD_ALTER_GEOM_FIELD_DEFN_FLAGS "DMD_ALTER_GEOM_FIELD_DEFN_FLAGS"
 
 #define GDAL_DCAP_OPEN       "DCAP_OPEN"
 #define GDAL_DCAP_CREATE     "DCAP_CREATE"
+#define GDAL_DCAP_CREATE_MULTIDIMENSIONAL "DCAP_CREATE_MULTIDIMENSIONAL"
```

Comment:
> Perhaps we should go for the constants without GDAL_ for new stuff, and offer the shorter aliases for the existing ones ?

Sounds good. We could always just add aliases for ALL constants so that they all work with/without the prefix too...

Reply:
> We could always just add aliases for ALL constants so that they all work with/without the prefix too...

yes probably better

Reply:
Ok, I've gone with exposing both variants